src/QueryBuilder/ResultViewer/ResultViewer.js

```diff
@@ -0,0 +1,119 @@
+import React, { memo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Col, Row, Accordion, AccordionSet, MultiColumnList, Headline } from '@folio/stripes/components';
+import { PrevNextPagination, usePagination } from '@folio/stripes-acq-components';
+import { QueryLoader } from './QueryLoader';
+import { getTableMetadata } from './helpers';
+
+export const ResultViewer = memo(({
+  showPagination = true,
+  contentData = [],
+  limit = 100,
+  offset = 0,
+  onPageChange,
+  entityType,
+  headline,
+  headlineEnd,
+  visibleColumns,
+  onSetDefaultVisibleColumns,
+  onSetDefaultColumns,
+  accordionHeadline,
+  isInProgress,
+  inProgressTitle,
+  height,
+  mlcLoading,
+}) => {
+  const { changePage, pagination } = usePagination({ limit, offset });
+  const { columnMapping, defaultColumns, defaultVisibleColumns } = () => getTableMetadata(entityType);
+
+  useEffect(() => {
+    onPageChange?.(pagination);
+  }, [pagination]);
+
+  // set visible by default columns once
+  useEffect(() => {
+    if (defaultVisibleColumns && defaultColumns) {
+      onSetDefaultColumns?.(defaultColumns);
+      onSetDefaultVisibleColumns?.(defaultVisibleColumns);
+    }
+  }, [defaultVisibleColumns, defaultColumns]);
+
+  const renderTable = () => (
+    <>
+      <Row between="xs">
+        <Col xs={6}>
+          <Headline size="large" margin="none" tag="h3">
+            {headline}
+          </Headline>
+        </Col>
+        {headlineEnd}
+      </Row>
+      <Row>
+        <Col xs={12}>
+          <MultiColumnList
+            contentData={contentData}
+            columnMapping={columnMapping}
+            visibleColumns={visibleColumns}
+            pagingType={null}
+            onNeedMoreData={changePage}
+            height={height}
+            loading={mlcLoading}
+          />
+          {showPagination && (
+            <PrevNextPagination
+              {...pagination}
+              totalCount={contentData?.length}
+              onChange={changePage}
+            />
+          )}
+        </Col>
+      </Row>
+    </>
+  );
+
+  const renderWithAccordion = () => (
+    <AccordionSet>
```

Comment:
I don't suggest to wrap it in AccordionSet, in bulk-edit we'll have 2 accordions and they should be under the same accordion set